pkg/sql/user.go

```diff
@@ -473,9 +473,20 @@ func (p *planner) GetAllRoles(ctx context.Context) (map[username.SQLUsername]boo
 	return users, nil
 }
 
-// RoleExists returns true if the role exists.
+// RoleExists returns true if the role exists. If a role is found to exist,
+// the existence will be cached for the duration of the transaction.
 func (p *planner) RoleExists(ctx context.Context, role username.SQLUsername) (bool, error) {
-	return RoleExists(ctx, p.InternalSQLTxn(), role)
+	cache := p.EvalContext().RoleExistsCache
+	if cache != nil {
```

Comment:
What causes the cache to be nil here?

Reply:
it shouldn't be nil, it was just a defensive check

Reply:
IMO defensive checks like this make the code harder to maintain because it is hard to tell the difference between something that is expected to be nil and a bug, but I'll leave it up to you if you want to keep it.